modxcms/revolution
# Add new config option for number TVs

### What does it do?
Added a new boolean parameter and behavior via and override to Ext.form.NumberField to allow decimal place zero padding for number TVs. Note that this should have no side effects for current installations, as the default value for this new parameter is false (which maintains the current behavior).

### Why is it needed?
Currently it is not possible with the number TV type to save or display a number such as 52.00 or 3.10. Trailing zeros will always be trimmed (yielding 52 and 3.1 respectively). While from a math/technical perspective these are the same value, it's not desirable to lose trailing zeros when working with currency values for example.

### How to test
Clear your caches and remember to run `grunt build` within the __build/templates/default_ directory to compile the changes into the main js file. Create a few TVs with different setting for Allow Decimals, Decimal Precision, and the new parameter Strict Decimal Precision. Create a resource with access to these test TVs and play with saving different values. Observe that these values persist in the editing form and display as expected on the rendered web page.

### Related issue(s)/PR(s)
Provides the requested feature in #15838.
